### PR TITLE
fix(Header): Fix `image` prop

### LINKED ISSUE #542: bug(Header): Invalid propTypes validation
We have bug with PropTypes in `header`.

> Warning: Failed prop type: `Header` prop `children` conflicts with props: `image`. They cannot be defined together, choose one or the other.

``` html
<h4 class="ui image header">
  <img src="/images/avatar2/small/lena.png" class="ui mini rounded image">
  <div class="content">
    Lena
    <div class="sub header">Human Resources</div>
  </div>
</h4>
```

``` jsx
<Header as="h4" image>
  <Image src="http://semantic-ui.com/images/avatar2/small/lena.png" shape='rounded' size='mini'/>
  <Header.Content>
    Lena
    <Header.Subheader>Human Resources</Header.Subheader>
  </Header.Content>
</Header>
```

We can fix this with:

``` diff
image: customPropTypes.every([
-  customPropTypes.disallow(['children', 'icon']),
+  customPropTypes.disallow(['icon']),
-  PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.element,
-    PropTypes.object,
+  customPropTypes.givenProps(
+    { children: PropTypes.node.isRequired },
+    PropTypes.bool,
+  ),
+  customPropTypes.givenProps(
+    { image: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.object]) },
+    customPropTypes.disallow(['children']),
+  ),
  ]),
]),
```

I'll make PR?

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #542.

### CODE REVIEW

File: src/elements/Header/Header.js [hunk 1]
```diff
@@ -53,7 +54,7 @@ function Header(props) {
     )
   }
 
-  if (image || icon && typeof icon !== 'boolean') {
+  if ((image && typeof image !== 'boolean') || (icon && typeof icon !== 'boolean')) {
```
Comment (layershifter) [inline src/elements/Header/Header.js:57]: Impoved condition.

File: test/specs/elements/Header/Header-test.js [hunk 2]
```diff
@@ -36,13 +36,16 @@ describe('Header', () => {
...
+    it('adds an image class when true', () => {
+      shallow(<Header image />)
+        .should.have.className('image')
     })
-    it('does not add an icon class given a name', () => {
-      shallow(<Header icon='user' />)
-        .should.not.have.className('icon')
+    it('does not add an Image when true', () => {
+      shallow(<Header image />)
+        .should.not.have.descendants('Image')
```
Comment (layershifter) [inline test/specs/elements/Header/Header-test.js:53]: Added test.

File: test/specs/elements/Header/Header-test.js [hunk 3]
```diff
@@ -36,13 +36,16 @@ describe('Header', () => {
...
+  })
+
+  describe('image', () => {
+    it('adds an image class when true', () => {
+      shallow(<Header image />)
+        .should.have.className('image')
     })
-    it('does not add an icon class given a name', () => {
-      shallow(<Header icon='user' />)
-        .should.not.have.className('icon')
```
Comment (layershifter) [inline test/specs/elements/Header/Header-test.js]: Removed, it's covered by `common.implementsIconProp()`.

File: src/elements/Header/Header.js [hunk 4]
```diff
@@ -29,14 +29,15 @@ function Header(props) {
...
     useKeyOrValueAndKey(attached, 'attached'),
+    useKeyOnly(block, 'block'),
+    useKeyOnly(disabled, 'disabled'),
+    useKeyOnly(dividing, 'dividing'),
     useValueAndKey(floated, 'floated'),
+    useKeyOnly(icon === true, 'icon'),
+    useKeyOnly(image === true, 'image'),
     useKeyOnly(inverted, 'inverted'),
-    useKeyOnly(disabled, 'disabled'),
+    useKeyOnly(sub, 'sub'),
```
Comment (layershifter) [inline src/elements/Header/Header.js:40]: Sorted props.

File: test/specs/elements/Header/Header-test.js [hunk 5]
```diff
@@ -69,7 +80,14 @@ describe('Header', () => {
   })
 
   describe('subheader', () => {
-    it('adds HeaderSubheader as child when there is an icon', () => {
+    it('adds HeaderSubheader as child', () => {
```
Comment (levithomason) [inline test/specs/elements/Header/Header-test.js]: This one should be covered by `implementsShorthandProp()` for `subheader`.  Should be able to remove it without a change in coverage.

**Review by levithomason — CHANGES_REQUESTED** [review]
Just one comment, then good to merge.

File: test/specs/elements/Header/Header-test.js (hunk 5 above)
Comment (layershifter) [reply test/specs/elements/Header/Header-test.js]: Looks like issue with merge, now okay.

**Review by levithomason — APPROVED** [review]

**Comment by levithomason** [comment]
Released in `stardust@0.46.2`, thanks!

**Pull request merged**